src/context/toastNotificationProvider.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  NotificationAction,
+  NotificationType,
+  ValueOf,
+  failure,
+  info,
+  success,
+} from "@canonical/react-components";
+import { NotificationSeverity } from "@canonical/react-components/dist/components/Notification/Notification";
+import ToastNotification from "components/ToastNotification";
+import ToastNotificationList from "components/ToastNotificationList";
+import React, {
+  FC,
+  PropsWithChildren,
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+const HIDE_NOTIFICATION_DELAY = 5_000;
+
+export type ToastNotificationType = NotificationType & {
+  timestamp?: ReactNode;
+  id: string;
+};
+
+type ToastNotificationHelper = {
+  notifications: ToastNotificationType[];
+  success: (message: ReactNode) => ToastNotificationType;
+  info: (message: ReactNode, title?: string) => ToastNotificationType;
+  failure: (
+    title: string,
+    error: unknown,
+    message?: ReactNode,
+    actions?: NotificationAction[],
+  ) => ToastNotificationType;
+  clear: (notification?: ToastNotificationType[]) => void;
+  toggleListView: () => void;
+  isListView: boolean;
+  countBySeverity: GroupedNotificationCount;
+};
+
+export type GroupedNotificationCount = {
+  [key in ValueOf<typeof NotificationSeverity>]?: number;
+};
+
+const initialNotification: ToastNotificationType = {
+  id: "",
+  message: "",
+  type: "positive",
+};
+const ToastNotificationContext = createContext<ToastNotificationHelper>({
+  notifications: [],
+  success: () => initialNotification,
+  info: () => initialNotification,
+  failure: () => initialNotification,
+  clear: () => null,
+  toggleListView: () => null,
+  isListView: false,
+  countBySeverity: {},
+});
+
+interface Props {}
+
+const ToastNotificationProvider: FC<PropsWithChildren<Props>> = ({
```

Comment:
```suggestion
const ToastNotificationProvider: FC<PropsWithChildren> = ({ children }) => {
```
No need for an empty interface.